pkg/util/loadbalancer/loadbalancer.go

```diff
@@ -0,0 +1,30 @@
+package loadbalancer
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"strings"
+
+	mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-08-01/network"
+)
+
+func RemoveFrontendIPConfiguration(lb *mgmtnetwork.LoadBalancer, resourceID string) error {
+	newFrontendIPConfig := make([]mgmtnetwork.FrontendIPConfiguration, 0, len(*lb.FrontendIPConfigurations))
+	for _, fipConfig := range *lb.LoadBalancerPropertiesFormat.FrontendIPConfigurations {
```

Comment:
With the current flow, there is a GET against the resource ID of the frontend IP Configuration, so the pointer should never be nil at this point.

Reply:
Though it doesn't prevent other calling code from not passing in populated data, but IMO if there is a loadbalancer that doesn't have frontend IP configuration there is a bigger problem.

Reply:
That's true, but I feel we should keep in mind that even if we can't control what Azure's APIs are returning to us we can control how we respond. I agree that missing data here would definitely be an indicator of a bigger problem, but that theoretical problem will be easier to handle in prod if we error out elegantly instead of `panic`-ing.

Reply:
@kimorris27 & @AldoFusterTurpin I'm addressing this in a follow up PR `https://github.com/Azure/ARO-RP/pull/3157`.  There are other public functions that are added that assume the pointers are safe but with the concerns raised here I'll make sure to check for nil where appropriate.